File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/data/Schedule.java
package com.non_name_hero.calenderview.data;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.Date;

@Entity(tableName = "schedule")
public class Schedule {

    @PrimaryKey(autoGenerate = true)
    @NonNull
    @ColumnInfo(name = "schedule_id")
    private long mScheduleId;

    @Nullable
    @ColumnInfo(name = "title")
    private String mTitle;

    @Nullable
    @ColumnInfo(name = "description")
    private String mDescription;

    @NonNull
    @ColumnInfo(name = "start_at_datetime")
    private Date mStartAtDatetime;

    @Nullable
    @ColumnInfo(name = "end_at_datetime")
    private Date mEndAtDatetime;

    @Nullable
    @ColumnInfo(name = "payment_id")
    private int mPaymentId;

    @Nullable
    @ColumnInfo(name = "group_id")
    private int mGroupId;

    @ColumnInfo(name = "start_timestamp")
    private String mStartTimestamp;

    @ColumnInfo(name = "end_timestamp")
    private String mEndTimestamp;

    //DBでは管理されないフィールド
    @Ignore
    private boolean mEditable;

    public Schedule(@NonNull long scheduleId,
                    @NonNull String title,
                    @NonNull String description,
                    @NonNull Date startAtDatetime,
                    @Nullable Date endAtDatetime,
                    @Nullable int groupId,
                    @Nullable int paymentId) {
        mScheduleId = scheduleId;
        mTitle = title;
        mDescription = description;
        mStartAtDatetime = startAtDatetime;
        mEndAtDatetime = endAtDatetime;
        mPaymentId = paymentId;
        mGroupId = groupId;
        mEditable = true;
    }

    @Ignore
    public Schedule(@NonNull String title,
                    @NonNull String description,
                    @NonNull Date startAtDatetime,
                    @Nullable Date endAtDatetime,
                    @Nullable int groupId,
                    @Nullable int paymentId) {
        this(0, title, description, startAtDatetime, endAtDatetime, groupId, paymentId);
    }
    @Ignore
    public Schedule(@NonNull String title,
                    @Nullable Date datetime) {
        this(0, title, "", datetime, datetime, 0, 0);
    }


//    /**
//     * Use this constructor to create a new active schedule.
//     *
//     * @param title       title of the task
//     * @param description description of the task
//     */
//    @Ignore
//    public Schedule(@Nullable String title, @Nullable String description) {
//        this(title, description, UUID.randomUUID().toString(), false);
//    }

    @NonNull
    public long getScheduleId() {
        return mScheduleId;
    }

    public void setScheduleId(@Nullable long scheduleId) {
        this.mScheduleId = scheduleId;
    }

    @Nullable
    public String getDescription() {
        return mDescription;
    }

    public void setDescription(@Nullable String mDescription) {
        this.mDescription = mDescription;
    }

    @Nullable
    public String getTitle() {
        return mTitle;
    }

    public void setTitle(@Nullable String mTitle) {
        this.mTitle = mTitle;
    }

    @Nullable
    public Date getStartAtDatetime() {
        return mStartAtDatetime;
    }

    public void setStartAtDatetime(@NonNull Date mStartAtDatetime) {
        this.mStartAtDatetime = mStartAtDatetime;
    }

    @Nullable
    public Date getEndAtDatetime() {
        return mEndAtDatetime;
    }

    public void setEndAtDatetime(@Nullable Date mEndAtDatetime) {
        this.mEndAtDatetime = mEndAtDatetime;
    }

    @Nullable
    public int getGroupId() {
        return mGroupId;
    }

    public void setGroupId(int mGroupId) {
        this.mGroupId = mGroupId;
    }

    @Nullable
    public int getPaymentId() {
        return mPaymentId;
    }

    public void setPaymentId(int mPaymentId) {
        this.mPaymentId = mPaymentId;
    }

    public String getStartTimestamp() {
        return mStartTimestamp;
    }

    public void setStartTimestamp(String startTimestamp) {
        mStartTimestamp = startTimestamp;
    }

    public String getEndTimestamp() {
        return mEndTimestamp;
    }

    public void setEndTimestamp(String startTimestamp) {
        mEndTimestamp = startTimestamp;
    }

    public void setUneditable() {
        mEditable = false;
    }
}

File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/data/Schedule.kt
package com.non_name_hero.calenderview.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.non_name_hero.calenderview.utils.PigLeadUtils
import java.util.*

@Entity(tableName = "schedule")
class Schedule(
    @field:ColumnInfo(name = "schedule_id") @field:PrimaryKey(autoGenerate = true) var scheduleId: Long,
    title: String,
    description: String,
    startAtDatetime: Date,
    endAtDatetime: Date?,
    groupId: Int,
    timeSettingFlag: Boolean
) {

    @ColumnInfo(name = "title")
    var title: String?

    @ColumnInfo(name = "description")
    var description: String?

    @ColumnInfo(name = "start_at_datetime")
    private var mStartAtDatetime: Date

    @ColumnInfo(name = "end_at_datetime")
    var endAtDatetime: Date?

    @ColumnInfo(name = "time_setting_flag")
    var timeSettingFlag: Boolean

    @ColumnInfo(name = "group_id", defaultValue = "1")
    var groupId: Int

    @ColumnInfo(name = "start_timestamp")
    var startTimestamp: String = ""

    @ColumnInfo(name = "end_timestamp")
    var endTimestamp: String = ""

    //DBでは管理されないフィールド
    @Ignore
    private var mEditable: Boolean

    @Ignore
    var isHoliday = false

    @Ignore
    constructor(
        title: String,
        description: String,
        startAtDatetime: Date,
        endAtDatetime: Date?,
        groupId: Int,
        timeSettingFlag: Boolean
    ) : this(0, title, description, startAtDatetime, endAtDatetime, groupId, timeSettingFlag) {
    }

    val startAtDatetime: Date?
        get() = mStartAtDatetime

    fun setStartAtDatetime(mStartAtDatetime: Date) {
        this.mStartAtDatetime = mStartAtDatetime
    }

    fun setUneditable() {
        mEditable = false
    }

    init {
        this.title = title
        this.description = description
        mStartAtDatetime = startAtDatetime
        this.endAtDatetime = endAtDatetime
        this.timeSettingFlag = timeSettingFlag
        this.groupId = groupId
        this.startTimestamp = PigLeadUtils.formatYYYYMMDD.format(startAtDatetime)
        this.endTimestamp = PigLeadUtils.formatYYYYMMDD.format(endAtDatetime)
        mEditable = true
    }
}